src/common.browser/WebsocketMessageAdapter.ts

```diff
@@ -123,34 +126,37 @@ export class WebsocketMessageAdapter {
-                        const tlsSocket: tls.TLSSocket = e.socket as tls.TLSSocket;
-                        const peer: tls.DetailedPeerCertificate = tlsSocket.getPeerCertificate(true);
-
-                        // Cork the socket until we know if the cert is good.
-                        tlsSocket.cork();
-
-                        ocsp.check({
-                            cert: peer.raw,
-                            httpOptions: httpsOptions,
-                            issuer: peer.issuerCertificate.raw,
-                        }, (error: Error, res: any): void => {
-                            if (error) {
-                                this.privCertificateValidatedDeferral.reject(error.message);
-                                tlsSocket.destroy(error);
-                            } else {
-                                this.privCertificateValidatedDeferral.resolve(true);
-                                tlsSocket.uncork();
-                            }
+                    if (enableOCSP) {
+                        this.privWebsocketClient.addListener("upgrade", (e: http.IncomingMessage): void => {
+                            const tlsSocket: tls.TLSSocket = e.socket as tls.TLSSocket;
+                            const peer: tls.DetailedPeerCertificate = tlsSocket.getPeerCertificate(true);
+
+                            // Cork the socket until we know if the cert is good.
+                            tlsSocket.cork();
+
+                            ocsp.check({
+                                cert: peer.raw,
+                                httpOptions: httpsOptions,
+                                issuer: peer.issuerCertificate.raw,
+                            }, (error: Error, res: any): void => {
+                                if (error) {
+                                    this.privCertificateValidatedDeferral.reject(error.message);
+                                    tlsSocket.destroy(error);
+                                } else {
+                                    this.privCertificateValidatedDeferral.resolve(true);
+                                    tlsSocket.uncork();
+                                }
+                            });
                         });
-                    });
+                    }
 
                 } else {
-                    // The ocsp library will handle validation for us and fail the connection if needed.
-                    this.privCertificateValidatedDeferral.resolve(true);
-
-                    const ocspAgent: ocsp.Agent = new ocsp.Agent({});
-                    const options: ws.ClientOptions = { agent: ocspAgent };
+                    const options: ws.ClientOptions = {};
+                    if (enableOCSP) {
+                        // The ocsp library will handle validation for us and fail the connection if needed.
+                        this.privCertificateValidatedDeferral.resolve(true);
```

Comment:
Ran the test suite with NODE_TLS_REJECT_UNAUTHORIZED = 0 and all the tests that use the service failed. 

I suspect it's due to this deferral never being resulted if enable OCSP is false. For the non-proxy case it should always be resolved, and in the proxy case resolved if enableOCSP is false.